lib/main_publiccloud.pm

```diff
@@ -221,6 +221,10 @@ sub load_publiccloud_tests {
             load_publiccloud_download_repos();
         } elsif (get_var('PUBLIC_CLOUD_QAM')) {
             load_maintenance_publiccloud_tests();
+        } elsif (get_var('PUBLIC_CLOUD_AISTACK')) {
+            my $args = OpenQA::Test::RunArgs->new();
```

Comment:
This is a duplication of line 215. Maybe move it to the top of the subroutine like `load_maintenance_publiccloud_tests` and `load_latest_publiccloud_tests`?

Reply:
I think `new` is not-duplicate it is local to the if-branch used to 
 pass args between those routines call only
Ok as it is.

Reply:
217: `my $args = OpenQA::Test::RunArgs->new();`
225: `my $args = OpenQA::Test::RunArgs->new();`

They will both evaluate to the same result, no? The other subroutines that I mentioned have a single line like that, at the top of the scope.

Reply:
Your idea still is valid. 
But this routine has never-sequential modules in each if-branch and  we have 2 only branches using $args locally (never executed in sequence, mutually exclusive).
Therefore 2 spearated local parameters logic is also correct and more conservative of current status, minimizing changes.